server/runtime/edit_tool_executor_test.go

```
@@ -0,0 +1,113 @@
+
+type capturingTool struct {
+	name toolspec.ID
+}
+
+func (t capturingTool) Name() toolspec.ID { return t.name }
+
+func (t capturingTool) Call(_ context.Context, c tools.Call) (tools.Result, error) {
+	var payload map[string]any
+	if err := json.Unmarshal(c.Input, &payload); err != nil {
+		out, _ := json.Marshal("Edit failed: expected JSON object input.")
+		return tools.Result{CallID: c.ID, Name: c.Name, Output: out, IsError: true, Summary: "Edit failed: expected JSON object input."}, nil
```

golangci-lint (2.12.1)

[error] error is not nil (line 107) but it returns nil

(nilerr)